components/pd_client/src/client_v2.rs

```diff
@@ -0,0 +1,1339 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! PD Client V2
+//!
+//! In V1, the connection to PD and related states are all shared under a
+//! `RwLock`. The maintenance of these states are implemented in a
+//! decentralized way: each request will try to rebuild the connection on its
+//! own if it encounters a network error.
+//!
+//! In V2, the responsibility to maintain the connection is moved into one
+//! single long-running coroutine, namely [`reconnect_loop`]. Users of the
+//! connection subscribe changes instead of altering it themselves.
+
+use std::{
+    collections::HashMap,
+    fmt::Debug,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, Mutex as StdMutex,
+    },
+    time::Duration,
+    u64,
+};
+
+use fail::fail_point;
+use futures::{
+    channel::mpsc,
```

Comment:
Why not use tikv_util's version? For example, how to achieve batch using futures' version?

Reply:
Okay, but for batching is it equivalent to futures' [ready_chunks](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.ready_chunks)?

Reply:
They are not the same. `tikv_util` will queue the items without notifying receiver, `ready_chunks` will always notify receiver, but receiver will try batch as much as possible. The former does the batch on sender side, the later does batch on receiver side. TSO can't do batching with the returned stream/sink unless batch capacity is passed as parameter. Batch on sender side is good for throughput technically, as it allows precise control on when a batch is finished.